gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -136,6 +146,45 @@ public WritableByteChannel create(StorageResourceId resourceId, CreateObjectOpti
     return channel;
   }
 
+  /**
+   * See {@link GoogleCloudStorage#createBucket(String, CreateBucketOptions)} for details about
+   * expected behavior.
+   */
+  @Override
+  public void createBucket(String bucketName, CreateBucketOptions options) throws IOException {
+    logger.atFiner().log("createBucket(%s)", bucketName);
+    checkArgument(!isNullOrEmpty(bucketName), "bucketName must not be null or empty");
+    checkNotNull(options, "options must not be null");
+    checkNotNull(storageOptions.getProjectId(), "projectId must not be null");
```

Comment:
do we have this check for other APIs?

Reply:
No, this is required only for create and list bucket APIs. We have similar check for HTTP client https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/48891bff6220a309810307753a0f413268d3aba2/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java#L452